core/src/main/java/org/apache/accumulo/core/data/Key.java

```diff
@@ -185,6 +185,44 @@ public Key(byte row[], int rOff, int rLen, byte cf[], int cfOff, int cfLen, byte
   }
 
   /**
+   * Creates a key.
+   *
+   * @param row
+   *          bytes containing row ID
+   * @param rOff
+   *          offset into row where key's row ID begins (inclusive)
+   * @param rLen
+   *          length of row ID in row
+   * @param cf
+   *          bytes containing column family
+   * @param cfOff
+   *          offset into cf where key's column family begins (inclusive)
+   * @param cfLen
+   *          length of column family in cf
+   * @param cq
+   *          bytes containing column qualifier
+   * @param cqOff
+   *          offset into cq where key's column qualifier begins (inclusive)
+   * @param cqLen
+   *          length of column qualifier in cq
+   * @param cv
+   *          bytes containing column visibility
+   * @param cvOff
+   *          offset into cv where key's column visibility begins (inclusive)
+   * @param cvLen
+   *          length of column visibility in cv
+   * @param ts
+   *          timestamp
+   * @param deleted
+   *          delete marker
+   * @param copy
+   *          if true, forces copy of byte array values into key
+   */
+  public Key(byte row[], int rOff, int rLen, byte cf[], int cfOff, int cfLen, byte cq[], int cqOff, int cqLen, byte cv[], int cvOff, int cvLen, long ts, boolean deleted, boolean copy) {
```

Comment:
Because it's new and something that we'd have to support forever. It has 15 arguments which is a user error waiting to happen. If I could go back in time and drop those other constructors in favor of a builder pattern, I probably would. The whole point of what we're doing here is to avoid exposing the additional combinatorial explosion of parameters.